slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/BlockImporter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.interchange;
+
+import tech.pegasys.web3signer.slashingprotection.dao.LowWatermarkDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlocksDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SigningWatermark;
+import tech.pegasys.web3signer.slashingprotection.dao.Validator;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock;
+import tech.pegasys.web3signer.slashingprotection.validator.BlockValidator;
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.jdbi.v3.core.Handle;
+
+public class BlockImporter {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  final OptionalMinValueTracker minSlotTracker = new OptionalMinValueTracker();
+
+  private final LowWatermarkDao lowWatermarkDao;
+  private final SignedBlocksDao signedBlocksDao;
+
+  private final Validator validator;
+  private final Handle handle;
+  private final ObjectMapper mapper;
+
+  public BlockImporter(
+      final Validator validator,
+      final Handle handle,
+      final ObjectMapper mapper,
+      final LowWatermarkDao lowWatermarkDao,
+      final SignedBlocksDao signedBlocksDao) {
+    this.validator = validator;
+    this.handle = handle;
+    this.mapper = mapper;
+    this.lowWatermarkDao = lowWatermarkDao;
+    this.signedBlocksDao = signedBlocksDao;
+  }
+
+  public void importFrom(final ArrayNode signedBlocksNode) throws JsonProcessingException {
+
+    for (int i = 0; i < signedBlocksNode.size(); i++) {
+      final SignedBlock jsonBlock = mapper.treeToValue(signedBlocksNode.get(i), SignedBlock.class);
+      final BlockValidator blockValidator =
+          new BlockValidator(
+              handle,
+              jsonBlock.getSigningRoot(),
+              jsonBlock.getSlot(),
+              validator.getId(),
+              signedBlocksDao,
+              lowWatermarkDao);
+
+      if (jsonBlock.getSigningRoot() == null) {
```

Comment:
same suggestion as in the attestationImporter could combine the two conditions in the same if expression

Reply:
not sure that we can